cypress-tests/cypress/e2e/admin/pageBuilder/templates/templatePreview.cy.ts

```diff
@@ -0,0 +1,64 @@
+import { customAlphabet } from "nanoid";
+
+context("Page Builder - Blocks", () => {
+    const nanoid = customAlphabet("abcdefghijklmnopqrstuvwxyz");
+    const pageTemplateData1 = {
+        title: nanoid(6),
+        slug: nanoid(6),
+        description: nanoid(6),
+        tags: [],
+        layout: "static",
+        pageCategory: "static"
+    };
+    const pageTemplateData2 = {
+        title: nanoid(6),
+        slug: nanoid(6),
+        description: nanoid(6),
+        tags: [],
+        layout: "static",
+        pageCategory: "static"
+    };
+    const pageTemplateData3 = {
+        title: nanoid(6),
+        slug: nanoid(6),
+        description: nanoid(6),
+        tags: [],
+        layout: "static",
+        pageCategory: "static"
+    };
+
+    beforeEach(() => {
+        cy.login();
+        cy.pbDeleteAllTemplates();
+        cy.pbCreatePageTemplate(pageTemplateData1);
+        cy.pbCreatePageTemplate(pageTemplateData2);
+        cy.pbCreatePageTemplate(pageTemplateData3);
+    });
+
+    it("Should be able to create a page and view all existing templates in it", () => {
+        cy.visit("/page-builder/pages?folderId=root");
+        cy.wait(1000);
+        cy.findByTestId("new-page-button").click();
+        cy.wait(500);
+        cy.contains("Pick a template for your new page").should("exist");
+        cy.contains(pageTemplateData1.title).should("exist");
+        cy.contains(pageTemplateData1.description).should("exist");
+        cy.contains(pageTemplateData2.title).should("exist");
+        cy.contains(pageTemplateData2.description).should("exist");
+        cy.contains(pageTemplateData3.title).should("exist");
+        cy.contains(pageTemplateData3.description).should("exist");
+
+        cy.get(".css-1rl9ll7-listStyle .css-5bicyh-listItem").each(item => {
+            // Click on the current item
+            cy.wrap(item).click();
+
+            // Wait for the right panel to load (adjust this timeout if needed)
+            cy.get(".webiny-split-view__right-panel").should("be.visible", { timeout: 10000 });
+            cy.contains(
```

Comment:
I don't understand this assertion here. Is this correct? What's the intention here? Can we simplify via 3 separate cy.contains calls?

Reply:
I thought this was a very nice solution. What happens is we click on each item on the left side of the screen one at a time and for each click we assert on the right side do we see the title AND description being loaded properly. And as you can see we have OR's inbetween the name-description pairs. This is so that I avoid using for loops and other ways of knowing what is being clicked.

This way the code clicks the first item on the list, asserts pageTemplateData1.title && pageTemplateData1.description
is happening and then clicks the second item and does the same. I think the current way might be the simplest I'll add comments and if you don't like this solution I can make it a bit more readable.